plugins/arc/src/ArcRenderer/ArcRendering.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react'
+import { readConfObject } from '@jbrowse/core/configuration'
+import { bpSpanPx, measureText } from '@jbrowse/core/util'
+import { observer } from 'mobx-react'
+import { Tooltip } from 'react-svg-tooltip'
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function ArcRendering(props: any) {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const onClick = (event: any, id: any) => {
+    const { onFeatureClick: handler } = props
+    if (!handler) {
+      return undefined
+    }
+    return handler(event, id)
+  }
+
+  const {
+    features,
+    config,
+    regions,
+    blockKey,
+    bpPerPx,
+    displayModel: { selectedFeatureId },
+  } = props
+  const [region] = regions
+  const arcsRendered = []
+
+  for (const feature of features.values()) {
+    const [left, right] = bpSpanPx(
+      feature.get('start'),
+      feature.get('end'),
+      region,
+      bpPerPx,
+    )
+
+    const featureId = feature.id()
+    const id = blockKey + '-' + featureId
+    let stroke = readConfObject(config, 'color', { feature })
+    let textStroke = 'black'
+    if (
+      selectedFeatureId &&
+      String(selectedFeatureId) === String(feature.id())
+    ) {
+      stroke = textStroke = 'red'
+    }
+    const label = readConfObject(config, 'label', { feature })
+    const caption = readConfObject(config, 'caption', { feature })
+    const strokeWidth = readConfObject(config, 'thickness', { feature }) || 1
+    const ref = React.createRef<SVGPathElement>()
+    const tooltipWidth = 20 + measureText(caption?.toString())
+
+    arcsRendered.push(
+      <g key={id} onClick={e => onClick(e, featureId)}>
+        <path
+          id={id}
+          d={`M ${left} 0 C ${left} 100, ${right} 100, ${right} 0`}
+          stroke={stroke}
+          strokeWidth={strokeWidth}
+          fill="transparent"
+          onClick={e => onClick(e, featureId)}
+          ref={ref}
+          pointerEvents="stroke"
+        />
+        <Tooltip triggerRef={ref}>
+          <rect
+            x={12}
+            y={0}
+            width={tooltipWidth}
+            height={20}
+            rx={5}
+            ry={5}
+            fill="black"
+            fillOpacity="50%"
+          />
+          <text
+            x={22}
+            y={14}
+            fontSize={10}
+            fill="white"
+            textLength={tooltipWidth - 20}
+          >
+            {caption}
+          </text>
+        </Tooltip>
+        <text
+          x={left + (right - left) / 2}
+          y="78px"
```

Comment:
is there any reasoning for the number 78px?

Reply:
the arcs were at a static height and this placed it at the apex i believe

with your change to the arc heights for clarity this would have to be updated and calculated based on the other properties of the arc.

no issue changing it if that's a part of the changes you proposed

Reply:
can see that it works but it is a bit of a magic number...wondering if it could be calculated(?) it is surprising a little bit that it is not just 100px since that is a control point on the bezier curve at least

Reply:
I think you can just do `y={height - 22}` (or - 21 looks a little more centred even) to calculate this y value for the text. I don't know enough about the bezier curve control point to generalize it based on the value we give it.

If the {height} value for the bezier curve is indeed the control point, _wouldn't_ that place it above the apex of the curve? The placement of the text is certainly exact pixels y value from its origin.

Reply:
Upon exploration, the equation for finding the apex/vertex of a bezier curve using the three points of the start, the end, and the control point is 
![image](https://user-images.githubusercontent.com/83305007/146076323-db16a3fb-a747-4899-954e-7bf18c05c929.png)

from https://math.stackexchange.com/questions/217522/how-do-you-find-the-vertex-of-a-b%C3%A9zier-quadratic-curve

math flashbacks

Reply:
using mathjs:

```
    const left = 0
    const right = 50
    const height = 100

    const p0 = [left, height]
    const p1 = [right, height]
    const p2 = [right, 0]

    const numerator = math.dot(math.subtract(p0, p1), math.subtract(math.add(p0, p2), math.multiply(2, p1)))
    
    const denominator = math.subtract(math.dot(math.add(p0, p2), math.add(p0, p2)), math.dot(math.multiply(4, p1), math.subtract(math.add(p0, p2), p1)))
    
    const t = math.divide(numerator, denominator)
    
    const textYCoord = math.add(t, math.multiply(height, t))
```

fiddle: https://jsbin.com/bilaxehafo/edit?html,output

...i think this is what you were looking for? might be way off base

Reply:
nice...if that works then might be valuable to incorporate.. could try with either using mathjs or without (might be useful to avoid a large bundle size increase since mathjs may be large-ish)

could also just try heuristics too, we may not need precise placement